lsp/source/grammarList.civet

```diff
@@ -0,0 +1,38 @@
+export features = [
+  {
+    key: 'coffeeComment'
+    marker: 'hash_comment' // The key for the repository
+    snippet: `{
+      "name": "comment.line.number-sign.civet",
+      "begin": "(?<=^|\\\\s)(?<!\\\\.)#(?!(\\\\{|##|\\\\s*/|\\\\s*\\\\]))",
+      "end": "$",
+      "beginCaptures": {
+        "0": {
+          "name": "punctuation.definition.comment.civet"
+        }
+      }
+    }`
+  }
+]
+
+###
+CoffeeComment:
+  (?<=^|\s)      - preceded by BOL or whitespace
+  (?<!\.)        - NOT preceded by a dot
+  #              - the hash itself
+  (?!(\{|##|\s*/|\s*\]))   - NOT followed by any of:
+                          -   {       -> interpolation '#{'
+                          -   ##      -> triple-hash block comment
+                          -   /       -> ‘divide-by’ operator
+                          -   ]       -> length token inside brackets
+
+Edge-case matrix covered:
+  # comment               + comment
+  code  # trailing        + comment
+  array.# length          X not comment
```

Comment:
I think this now parses as a comment after https://github.com/DanielXMoore/Civet/pull/1750